### PR TITLE
Use pdr sapphire wrapper for making transactions

### LINKED ISSUE #15: Use pdr-sapphire-wrapper for making transactions
Use pdr-sapphire-wrapper for making transactions

### PR DESCRIPTION
**Pull request by trizin**
Fixes #15

Changes proposed in this PR:

- Use pdr sapphire wrapper for making transactions

### CODE REVIEW

**Review by alexcos20 — CHANGES_REQUESTED** [review]
This will break all the existing flows, especially barge.

Proposal:
 - have a function called is_sapphire_rpc  (takes web3, returns true if chain_id is a match)
 - when needed in functions that require privacy, create the tx using encodeABI.
 - then check if is_sapphire_rpc  
 - if yes:  call wrapper.send_encrypted_sapphire_tx
 - else, sign the tx and use web3.py to send the transaction
 - keep wait_for_receipt check at the end

**Comment by trizin** [comment]
Right, I haven't thought about the impacts on local deployment. I'll update the PR, thank you! @alexcos20

File: pdr_utils/contract.py [hunk 1]
```diff
@@ -285,21 +327,70 @@ def soonest_block_to_predict(self, block):
...
+
         self.token.approve(self.contract_address, amount_wei)
         gasPrice = self.config.w3.eth.gas_price
         try:
-            tx = self.contract_instance.functions.submitPredval(
-                predicted_value, amount_wei, prediction_block
-            ).transact({"from": self.config.owner, "gasPrice": gasPrice})
-            print(f"Submitted prediction, txhash: {tx.hex()}")
+            txhash = None
+            if is_sapphire_rpc(self.config.rpc_url):
```
Comment (alexcos20) [inline pdr_utils/contract.py]: we should check based on chainId.
Maybe somebody is running their own node and then url is not a match

**Review by alexcos20 — CHANGES_REQUESTED** [review]

**Comment by alexcos20** [comment]
Mainnet Chain ID:    Hex: 0x5afe,Decimal: 23294
Testnet   Chain ID:    Hex: 0x5aff,Decimal: 23295

File: pdr_utils/contract.py (hunk 1 above)
Comment (trizin) [reply pdr_utils/contract.py]: I fixed it. Thanks!

**Review by alexcos20 — APPROVED** [review]

**Pull request merged**